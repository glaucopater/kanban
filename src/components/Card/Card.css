.Card {
  border: 1px solid black;
  background: tomato;
  width: 50%;
  margin: 25px auto;  
  color: white; 
  text-align: center;
}

.card-body {
  padding: 20px;
  word-break: break-all;
  text-align: center; 
}

.Card .ToDo {
  background: #2bb4ea;
}

.Card .InProgress {
  background: #f76176;
}

.Card .Done {
  background: #485e76;
}

.card-header {

}

.button-remove {  
  float: right;  
  background: transparent;  
  font-size: 1.2em;
  color: white;
  border: none;
}