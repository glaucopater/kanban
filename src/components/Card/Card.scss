@import "../../sass/variables.scss";
@import "../../sass/mixins.scss";

.Card {
  border: 1px solid $black;
  width: 50%;
  margin: 5 * $offsetWidth auto;  
  color: $white; 
  text-align: center;
  padding: 3 * $offsetWidth;
}

.card-body {
  padding: 4 * $offsetWidth;
  word-break: break-all;
  text-align: center; 
}

.ToDo .Card {
  background: $color-card-todo;
}

.InProgress .Card  {
  background: $color-card-inprogress;
}

.Done .Card {
  background: $color-card-done;
}

.button-remove {  
  @include button();
}