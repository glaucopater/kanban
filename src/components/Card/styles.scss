@import "../../styles/sass/variables.scss";
@import "../../styles/sass/mixins.scss";

.Card {
  width: 50%;
  margin: 5 * $offsetWidth auto;
  color: $white;
  text-align: center;
  position: relative;
  @include box-shadow(
    0 $offsetWidth 3 * $offsetWidth -$offsetWidth rgba($lightgrey, 0.1)
  );
}

.Card:hover {
  cursor: pointer;
}

.isDragged, .isDragged:hover {
  cursor: move;
  border: 1px dashed grey;
}

.card-header {
  outline: 1px solid $blue;
}

.card-body {
  padding: 2 * $offsetWidth;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
}

.ToDo .Card {
  background: $color-card-todo;
}

.InProgress .Card {
  background: $color-card-inprogress;
}

.Done .Card {
  background: $color-card-done;
}

.button-remove {
  @include button();
  position: absolute;
  top: 0;
  right: 0;
  opacity: 0;
  &:hover {
    opacity: 0.7;
  }
}

textarea {
  overflow: hidden;
  width: 80%;
}
