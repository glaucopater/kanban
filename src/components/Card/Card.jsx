import React from "react";
import "./Card.scss";
import PropTypes from 'prop-types';

const Card = props => {

  const handleOnClick = () => {
    props.removeCard(props.id);
  }

  const handleOnDragStart = (ev,id) => { 
    ev.dataTransfer.setData("id", id);
    ev.dataTransfer.setData("sourceCategory", props.category);
  }

  return (
    <div className={"Card " + props.category} 
    onDragStart={e => handleOnDragStart(e, props.id)}  
    draggable  
    >
    <header className="card-header">
      <button className="button-remove" onClick={handleOnClick}>x</button>
    </header>
    <div className="card-body">
      {props.text}
    </div>
  </div>)
  
};

export default Card;

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string
}