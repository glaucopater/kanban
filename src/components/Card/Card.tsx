import React from "react";
import "./Card.scss";

interface ICardProps {
  category: string;
  category_tasks?: any;
  createCard?: any;
  removeCard?: any;
  updateCard?: any;
  editable?: boolean;
  text?: string;
  value?: string;
  id: number;
}

interface ICardState {
  id?: number;
  text?: string;
  value?: string;
  editable?: boolean;
}

export class Card extends React.Component<ICardProps, ICardState> {
  constructor(props: ICardProps) {
    super(props);
    this.state = { editable: false, value: "", text: "", id: 0 };
    this.handleOnEdit = this.handleOnEdit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleOnRemove = this.handleOnRemove.bind(this);
  }

  handleOnEdit = () => {
    this.setState({ editable: true });
  };

  handleOnRemove = () => {
    this.props.removeCard(this.props.id);
  };

  handleOnBlur = (event: { target: { value: string; }; }) => {
    const cleanedValue = event.target.value.trim();
    if (cleanedValue !== "") {
      this.setState({ editable: false, value: event.target.value });
      this.props.updateCard({ id: this.props.id, text: this.state.text });
    }
  };

  handleChange = (event: any) => {
    this.setState({ value: event.target.value });
  };

  handleOnDragStart = (ev: any, id: number) => {
    ev.dataTransfer.setData("id", id);
    ev.dataTransfer.setData("sourceCategory", this.props.category);
  };

  getInputField() {
    return (
      <textarea
        value={this.state.value}
        onChange={this.handleChange}
        onBlur={this.handleOnBlur}
      />
    );
  }

  render() {
    const content = this.state.editable
      ? this.getInputField()
      : this.props.text;
    return (
      <div
        className={"Card " + this.props.category}
        onDragStart={(e) => this.handleOnDragStart(e, this.props.id)}
        draggable
      >
        <button className="button-remove" onClick={this.handleOnRemove}>
          x
        </button>
        <div className="card-body" onClick={this.handleOnEdit}>
          {content}
        </div>
      </div>
    );
  }
}

export default Card;
