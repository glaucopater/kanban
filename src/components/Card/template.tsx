import React from "react";
import "./styles.scss";

interface ICardProps {
  category: string;
  category_tasks?: any;
  createCard?: any;
  removeCard?: any;
  updateCard?: any;
  editable?: boolean;
  text?: string;
  value?: string;
  id: number;
}

interface ICardState {
  id?: number;
  text?: string;
  value?: string;
  editable?: boolean;
}

export const Card : React.FC<ICardProps> = (props) => {
  const [state, setState] = React.useState<ICardState>(props);
  React.useEffect(() => {
    setState({ editable: false, value: "", text: "", id: 0 });
  }, [props.category]);

  const handleOnEdit = () => {
    setState({ editable: true });
  };

  const handleOnRemove = () => {
    props.removeCard(props.id);
  };

  const handleOnBlur = (event: { target: { value: string; }; }) => {
    const cleanedValue = event.target.value.trim();
    if (cleanedValue !== "") {
      setState({ editable: false, value: event.target.value });
      props.updateCard({ id: props.id, text: state.text });
    }
  };

  const handleChange = (event: any) => {
    setState({ value: event.target.value });
  };

  const handleOnDragStart = (ev: any, id: number) => {
    ev.dataTransfer.setData("id", id);
    ev.dataTransfer.setData("sourceCategory", props.category);
  };

  const getInputField = () =>{
    return (
      <textarea
        value={state.value}
        onChange={handleChange}
        onBlur={handleOnBlur}
      />
    );
  }

  const content = state.editable
    ? getInputField()
    : props.text;

  return (
    <div
      className={"Card " + props.category}
      onDragStart={(e) => handleOnDragStart(e, props.id)}
      draggable
    >
      <button className="button-remove" onClick={handleOnRemove}>
        x
      </button>
      <div className="card-body" onClick={handleOnEdit}>
        {content}
      </div>
    </div>
  );

}