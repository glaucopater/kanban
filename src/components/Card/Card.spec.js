import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
    const props = { id: 1 , text : 'abc'}; 
    let card = shallow(<Card {...props}/>);
    it('renders properly', () => {
        expect(card).toMatchSnapshot();
    });
});