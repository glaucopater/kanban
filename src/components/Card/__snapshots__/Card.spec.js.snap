// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id={1}
    text="abc"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="button-remove"
          onClick={[Function]}
        >
          x
        </button>,
        <div
          className="card-body"
          onClick={[Function]}
        >
          abc
        </div>,
      ],
      "className": "Card undefined",
      "draggable": true,
      "onDragStart": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "x",
          "className": "button-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "x",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "abc",
          "className": "card-body",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "abc",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="button-remove"
            onClick={[Function]}
          >
            x
          </button>,
          <div
            className="card-body"
            onClick={[Function]}
          >
            abc
          </div>,
        ],
        "className": "Card undefined",
        "draggable": true,
        "onDragStart": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "button-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "abc",
            "className": "card-body",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "abc",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
